src/treq/test/test_content.py

```diff
@@ -173,3 +184,21 @@ def test_text_content_default_encoding_no_header(self):
         self.protocol.connectionLost(Failure(ResponseDone()))
 
         self.assertEqual(self.successResultOf(d), u'\xa1')
+
+    def test_text_content_unicode_headers(self):
+        """
+        Header parsing is robust against unicode header names and values.
+        """
+        self.response.headers = Headers({
+            b'Content-Type': [
```

Comment:
See also https://tools.ietf.org/html/draft-ietf-httpbis-header-structure-00 which tries to document the current state of things, and mostly concludes "hahahahahaha :sob:".